gnocchi/indexer/sqlalchemy.py

```diff
@@ -619,7 +619,9 @@ def create_archive_policy(self, archive_policy):
     def list_archive_policy_rules(self):
         with self.facade.independent_reader() as session:
             return session.query(ArchivePolicyRule).order_by(
-                ArchivePolicyRule.metric_pattern.desc()).all()
+                ArchivePolicyRule.metric_pattern.desc(),
```

Comment:
I'm not sure of the idea.
This does not guarantee that the longest pattern will appear in the front. For example:
```
pattern1: abc.*
pattern2: abc.(*)
```
metric `abc.(xyz)` should match `pattern2`, but if `metric_pattern.desc()` is used, `pattern1` is matched.

Reply:
Haha, that's exactly why I'm "WTF this is a bad idea" since the beginning but I was not able to find a proper example. Thanks @Asu4ni.

However, this is completely unrelated to this patch as this PR does not change that.

But the original PR was trying to fix that problem by ordering by pattern name. Maybe it's better.

When do we remove this archive policy rule entirely, @chungg ? :)

Reply:
Ceilometer now allows to specify the policy in each sink, so I guess removing it is fine.
BTW, in case one don't see, I comment above that should we add this point(name ordering) to the document before this merged?

Reply:
Yeah we can't really **remove** it, it'd break the API, but that's worth noting anyway. We probably need to document it, right. I'll wait for @gordc feedback. :)

Reply:
i'm indifferent. i guess we could just sort by length after we query but *shrugs* does it matter. i'm ok with as is and documenting/deprecating.

as you folks mention, the rules will always be sort of ambiguous because of regex.